base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -1968,6 +1990,28 @@ impl LMDBDatabase {
         if let Some(output_hash) = lmdb_get::<_, HashOutput>(txn, &self.payref_to_output_index, payref.as_slice())? {
             self.fetch_mined_info_by_output_hash_in_txn(txn, &output_hash)
         } else {
+            if let MetadataValue::PayrefRebuildStatus(status) =
+                lmdb_get::<_, MetadataValue>(txn, &self.metadata_db, &MetadataKey::PayrefRebuildStatus.as_u32())?
+                    .unwrap_or(MetadataValue::PayrefRebuildStatus(PayrefRebuildStatus::default()))
+            {
+                return if status.is_rebuilt {
+                    Ok(MinedInfo {
+                        input: None,
+                        output: None,
+                    })
+                } else {
+                    trace!(target: LOG_TARGET, "Payref index is not completed yet, current status: {:?}", status);
+                    Err(ChainStorageError::PayRefIndexNotAvailable {
+                        current_height: status.last_rebuild_height.unwrap_or_default(),
+                        start_height: if let Some(metadata) = status.metadata_at_start {
+                            metadata.best_block_height()
+                        } else {
+                            0
+                        },
+                        target_height: self.fetch_chain_metadata()?.best_block_height(),
+                    })
+                };
+            }
```

Comment:
```suggestion
            if let MetadataValue::PayrefRebuildStatus(status) =
                lmdb_get::<_, MetadataValue>(txn, &self.metadata_db, &MetadataKey::PayrefRebuildStatus.as_u32())?
                    .unwrap_or(MetadataValue::PayrefRebuildStatus(PayrefRebuildStatus::default()))
            {
               if !status.is_rebuilt {
                  
                   trace!(target: LOG_TARGET, "Payref index is not completed yet, current status: {:?}", status);
                    Err(ChainStorageError::PayRefIndexNotAvailable {
                        current_height: status.last_rebuild_height.unwrap_or_default(),
                        start_height: if let Some(metadata) = status.metadata_at_start {
                            metadata.best_block_height()
                        } else {
                            0
                        },
                        target_height: self.fetch_chain_metadata()?.best_block_height(),
                    })
                };
            }
```
soo many nested ifs
this makes it harder to read